docker/build_scripts/finalize.sh

```diff
@@ -44,8 +44,20 @@ source $TOOLS_PATH/bin/activate
 # Install default packages
 pip install -U --require-hashes -r $MY_DIR/requirements3.9.txt
 # Install certifi and auditwheel
+
+if [ "${POLICY}" == "musllinux_1_1" ]; then
+    # On Alpine cmake needs cmake to build. Install it from apk to
+    # build a newer version.
+    apk add --no-cache --virtual .cmake-deps cmake openssl-dev
+fi
+
 pip install -U --require-hashes -r $MY_DIR/requirements-tools.txt
 
+if [ "${POLICY}" == "musllinux_1_1" ]; then
+    # Remove the apk version
+    apk del .cmake-deps
+fi
+
```

Comment:
I think this is what's currently messing up CI for ppc64le (which is already flaky on manylinux) & s390x.
One way to get this working is probably to rely on Alpine provided CMake and don't install the one in requirements-tools (at least for now. Once the image is available, this will allow to build https://github.com/scikit-build/cmake-python-distributions).

Reply:
Yeah the cmake build takes more than 10 minutes on my machine. I'm going to split `requirements-tools.in` and remove cmake in the musllinux version.

Reply:
I think I'll rework tool installation with pipx later this week. I was already thinking about it but this is the occasion to do so.

Reply:
@lkollar, I refactored the way tools are installed. This shall allow to do something like this for now:
```bash
for TOOL_PATH in $(find . -type f); do
	TOOL=$(basename ${TOOL_PATH})
	if [ "${TOOL}" == "cmake" ] && [ "${POLICY}" == "musllinux_1_1" ]; then
		# TODO remove this exception once https://github.com/scikit-build/cmake-python-distributions
		# provides a musllinux wheel
		apk add cmake
	else
		pipx install --pip-args="--require-hashes -r" ${TOOL}
	fi
done
```